src/components/legend/draw.js

```diff
@@ -30,26 +30,44 @@ var getLegendData = require('./get_legend_data');
 var style = require('./style');
 var helpers = require('./helpers');
 
-module.exports = function draw(gd) {
+module.exports = function draw(gd, opts) {
     var fullLayout = gd._fullLayout;
     var clipId = 'legend' + fullLayout._uid;
+    var layer;
 
-    if(!fullLayout._infolayer || !gd.calcdata) return;
+    // Check whether this is the main legend (ie. called without any opts)
```

Comment:
This is little hacky.
What if we pass proper options to draw legend.
In that case one may look into `opts._main` to figure out the mocked legend (for unified hover) from the actual legend.